### PR TITLE
fix: Auth API send refresh token issue fixed

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR fixes issue of `Send scope with refresh token` dropdown in Authenticated API datasource. The issue was that this dropdown has a default value of No, whenever we update this from No to Yes and save the datasource, now if we edit this datasource, the value in this dropdown should be Yes but it was going back to default No. This PR fixes that issue

**Root cause:**
The issue occurred because in PR https://github.com/appsmithorg/appsmith/pull/29393 we had set Auth API dropdown values to default ones, so irrespective of the value in datasource config, these dropdowns were getting set to default values.

**Solution:**
This PR updates the initial values, so that if this value is already set, initial value is read from dsConfig, otherwise it's set to default value

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/29941
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
  - Improved the conditional rendering logic in the data source editor for a more intuitive user interface based on the selected authentication method.

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx [hunk 1]
```diff
@@ -721,7 +721,7 @@
               "",
               false,
               "",
-              false,
+              _.get(authentication, "sendScopeWithRefreshToken") || false,
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx:724]: Resolved. Please check

File: app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx [hunk 2]
```diff
@@ -679,7 +679,7 @@
             "",
             false,
             "",
-            false,
+            _.get(formData.authentication, "isAuthorizationHeader") || false,
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx:682]: Resolved. Please check

File: app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx [hunk 3]
```diff
@@ -595,7 +595,7 @@ class DatasourceRestAPIEditor extends React.Component<Props> {
             "",
             false,
             "",
-            true,
+            _.get(formData.authentication, "isTokenHeader") || true,
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx:598]: Resolved. Please check

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**